### DISCUSSION TITLE
Move Shared Folders first?

### DISCUSSION THREAD

**Discussion by Xeroxxx** [007revad/Synology_app_mover#85] [Q&A] [answered]
Hello, 
can someone explain if I should move Shared Folders like music, video or photos prior running the script and moving app or afterwards?

Same question for synology drive and home? Change Drive and Control Panel Home Service to Volume2 first or afterwards?

My goal is:
Move everything to Vol2 (single X16 16TB)
Decomission Vol1 (5 x 4TB Ironwolf)
Create new SHR1 4Kn Volume 1 (4x X16 16TB)
Move everything from Vol2 to Vol1
Decomission Vol2
Add last drive to Vol1 SHR.

Any thoughts on this?

Why not replacing the drives?
- Vol1 is RAID5 not SHR
- New drives are 4Kn, Vol1 is 512e.

Why not go from Vol2 then?
- Cosmetics I know. I want it to be Vol1
- I want the drives in Slot1-5 and not 2-6.

I do have a backup of important data like /homes. Everything else would be replaceable just in case.

Thank you guys!

**Comment by 007revad** [comment]
Your plan is what I'd do. I also prefer to have volume 1 instead of volume 2.

While the packages are stopped, you can move all your shared folders. You can do this before or after moving the packages. 

Some packages have a database that can be moved in the package's settings. At the end of the script it will show you what to do, if the package you moved needs any extra steps.
- Container Manager (Docker), Cloud Station, Download Station, Note Station, Synology Drive, Virtual Machine Manager and MariaDB each need extra steps (which are outlined at the end of the script).

**Comment by Xeroxxx** [comment]
So basically migrate everything to Vol2, keep everything stopped, move Shared Folders to Vol2. Move back to Vol1 and start the packages?

Currently backing up everything important to a tape drive :D

Thank you so much for giving a detailed answer in this short time. Keep on you good work!

**Reply by 007revad** [reply, marked as answer]
I usually recommend the following. It gives you a chance to check that all the packages moved okay before deleting storage pool 1. 
1. Migrate everything to vol2.
2. Shut down NAS.
3. Remove vol1 drives.
4. Boot NAS and mute the beep.
5. Start your packages and check they are okay.
6. Shut down NAS.
7. Install the drives you removed in step 3.
8. Boot NAS.
9. Delete vol1 storage pool.

**Comment by Xeroxxx** [comment]
Sry to bother again. @007revad 

Any recommendation to move user homes?

Disable UserHomes, Move, Enable Userhomes?

**Reply by Xeroxxx** [reply]
BTW can confirm ContainerManager 24.0.2-1520 (Beta) works as well.

**Reply by 007revad** [reply]
You can move the homes shared folder in Control Panel (while it is enabled).
https://imgur.com/gallery/synology-move-shared-folder-wnsyasb

Nice to see Synology finally updating docker from v20 to v24. Though Asustor have already updated theirs to v25.

**Reply by Xeroxxx** [reply]
Thanks. Found out moving the home folder automatically changes the home service setting. Switching drives now :)
